core/src/main/java/org/apache/calcite/rel/metadata/MetadataHandlerMethods.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.rel.metadata;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+
+/**
+ * Utilities useful for obtaining handler methods from a {@link MetadataHandler} class.
+ */
+public class MetadataHandlerMethods {
+  /**
+   * Finds handler methods defined by a {@link MetadataHandler}. Static and synthetic methods
+   * are ignored.
+   *
+   * @param handlerClass the handler class to inspect
+   * @return handler methods
+   */
+  public static Method[] handlerMethods(Class<? extends MetadataHandler<?>> handlerClass) {
```

Comment:
How about we put this as a static method on MetadataHandler instead of a separate class?

Reply:
Moved the method to `MetadataHandler` class